test/distrib/ruby/run_distrib_test.sh

```diff
@@ -17,12 +17,26 @@ set -ex
 
 cd "$(dirname "$0")"
 
-ARCH=$1
-PLATFORM=$2
+function die {
+  echo "$1"
+  exit 1
+}
+
+ARCH="$1"
+PLATFORM="$2"
+PACKAGE_TYPE="$3"
+echo "$EXTERNAL_GIT_ROOT"
+GRPC_VERSION="$(ruby -e 'require ENV["EXTERNAL_GIT_ROOT"] + "/src/ruby/lib/grpc/version.rb"; puts GRPC::VERSION')"
+if [[ "$PACKAGE_TYPE" == "source" ]]; then
+  GEM_NAME="grpc-${GRPC_VERSION}.gem"
+else
+  [[ "$PACKAGE_TYPE" == "binary" ]] || die "unexpeced package type: $PACKAGE_TYPE"
+  GEM_NAME="grpc-${GRPC_VERSION}-${ARCH}-${PLATFORM}.gem"
+fi
 # Create an indexed local gem source with gRPC gems to test
 GEM_SOURCE=../../../gem_source
 mkdir -p "${GEM_SOURCE}/gems"
-cp "$EXTERNAL_GIT_ROOT"/input_artifacts/grpc-*"$ARCH-$PLATFORM".gem "${GEM_SOURCE}/gems"
+cp "${EXTERNAL_GIT_ROOT}/input_artifacts/${GEM_NAME}" "${GEM_SOURCE}/gems"
```

Comment:
So the source-only gem (the one above named `grpc-${GRPC_VERSION}.gem`) already exists (or, already built in a previous step)?

Reply:
Yep, it's built in the artifacts job that is chained to the distrib tests job as a pre-req, and passed to this job via this `EXTERNAL_GIT_ROOT/input_artifacts` directory.